src/hsmm_mvpy/models.py

```diff
@@ -868,26 +924,34 @@ def sliding_event(self, n_events=None, color='k', figsize=(12,3), verbose=True,
             else:
                 return ax
         else:
-            return lkhs_init, mags_init, pars_init
+            return lkhs_init, mags_init, times_init
         
     def estimate_single_event(self, magnitudes, parameters, parameters_to_fix, magnitudes_to_fix, maximization, cpus, max_iteration=1e2, tolerance=1e-4):
         filterwarnings('ignore', 'Convergence failed, estimation hitted the maximum ', )#will be the case but for a subset only hence ignore
         if cpus is None:
             cpus = self.cpus
-        if np.shape(magnitudes) == 2:
-            magnitudes = np.tile(magnitudes, (len(parameters), 1, 1))
-        with mp.Pool(processes=self.cpus) as pool:
-            estimates = pool.starmap(self.EM, 
-                zip(itertools.repeat(1), magnitudes, parameters, 
-                    itertools.repeat(maximization), itertools.repeat(magnitudes_to_fix), 
-                    itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration),
-                    itertools.repeat(tolerance), itertools.repeat(1)))
+        if cpus > 1:
+            if np.shape(magnitudes) == 2:
+                magnitudes = np.tile(magnitudes, (len(parameters), 1, 1))
+            with mp.Pool(processes=self.cpus) as pool:
+                estimates = pool.starmap(self.EM, 
+                    zip(itertools.repeat(1), magnitudes, parameters, 
+                        itertools.repeat(maximization), itertools.repeat(magnitudes_to_fix), 
+                        itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration),
+                        itertools.repeat(tolerance)))
+        else:
```

Comment:
Somehow your commits contains an old code. This if/else function is no longer necessary, we can just keep the content of the if loop and remove the if

Reply:
Without the if/else it doesn't run because you get nested pools.

Reply:
Indeed, I forgot about that